proposals/0672-tuist-react-native.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Tuist.io in React Native
+author:
+- Riccardo Cipolleschi
+date: 2023-06-19
+---
+
+# RFC0672: Tuist.io in React Native 
+
+## Summary
+
+Managing Xcodeproj files is error prone and time consuming. 
+
+The current proposal suggests to move away from manually managing `Xcodeproj` files in favor of a declarative description leveraging [Tuist.io](https://tuist.io) as tool.
+
+## Basic example
+
+You can have a look at it in action in this PR:
+
+- https://github.com/facebook/react-native/pull/37952
+
+The idea is to describe the project with a declarative approach using Swift. Tuist allows you to load the `Project.swift` file in an instance of Xcode, leveraging also code completion and type checking to ensure the correctness of the project.
+
+## Motivation
+
+Managing Xcodeproj files is error prone and time consuming. Those files are source of frustrations for teams using them and they have been reported as one of the main pain point when users have to migrate to a new version of React Native.
+
+They are also an old way to manage projects. Even Apple realized that they are not the best approach and started developing Swift Package Manager (SwiftPM) that replace the project for developing libraries.
+
+Moder approaches relies on declarative syntax to describe how the project should look like and tools that can generate a valid Xcodeproj starting from that description.
+
+There are several tools that we can use, but the proposed one is [Tuist.io](https://tuist.io).
+
+[Tuist.io](https://tuist.io) is OSS and currently maintained by several companies, among which Shopify, which is already a React Native partner.
```

Comment:
Just a note that I happen to be at Shopify and a maintainer of Tuist, but Tuist and Shopify are not officially affiliated.

We don't have a company directly maintaining the tool but there are lots of large companies using it, some of them also financially sponsor us.